midx.c

```diff
@@ -706,7 +685,7 @@ static size_t write_midx_object_offsets(struct hashfile *f, int large_offset_nee
 	for (i = 0; i < nr_objects; i++) {
 		struct pack_midx_entry *obj = list++;
 
-		hashwrite_be32(f, obj->pack_int_id);
+		hashwrite_be32(f, perm[obj->pack_int_id]);
```

Comment:
Cant' see it in the code here, but is there a guard that this object doesn't refer to an expired pack?
That is, maybe assert(perm[obj->pack_int_id] != UINT_MAX)   ??